### PR TITLE
delete code before backup

### PR DESCRIPTION
**Pull request by gidsg**
https://trello.com/c/DrVWMkcZ/2169-rename-default-branches-on-our-repos-from-master-to-main

If a previous job fails it can leave a non-empty folder which causes subsequent jobs to fail:
e.g.
```
git clone https://4c3123d80a3e5000b9ed2ae0556f2dfb026d17b4@github.com/alphagov/digitalmarketplace-runner digitalmarketplace-runner --mirror
fatal: destination path 'digitalmarketplace-runner' already exists and is not an empty directory.
```
To fix this we delete the folder in case it exists when the job starts.

### CODE REVIEW

File: job_definitions/code_backups.yml [hunk 1]
```diff
@@ -57,6 +57,7 @@
       - ansicolor
     builders:
       - shell: |
+          rm -rf {{ package }}
```
Comment (bjgill) [inline job_definitions/code_backups.yml:60]: From line 47 above:

```
          wipe-workspace: false
```

Would setting this to `true` have the same effect as the `rm -rf`?

**Review by bjgill — APPROVED** [review]
I assume this was the thing you were talking about that can be removed once we've migrated all our repos to `main`?

File: job_definitions/code_backups.yml (hunk 1 above)
Comment (gidsg) [reply job_definitions/code_backups.yml:60]: Ah, good point, I missed that flag. I can try and set it and see how it behaves on Jenkins

**Comment by gidsg** [comment]
@bjgill, no the temporary thing is running the job and then manually changing the default branch in CodeCommit. This should remain as it helps to have a clean state if it fails part-way for whatever reason.

File: job_definitions/code_backups.yml [hunk 2]
```diff
@@ -57,6 +58,7 @@
       - ansicolor
     builders:
       - shell: |
+          cd ${WORKSPACE}
```
Comment (bjgill) [inline job_definitions/code_backups.yml:61]: Why is this necessary?

Comment (gidsg) [reply job_definitions/code_backups.yml:61]: Sorry, I pushed by accident whilst I was testing different approaches. I will force push once I have a better solution.

Comment (bjgill) [reply job_definitions/code_backups.yml:61]: OK - let me know when you want me to take a proper look.

**Comment by gidsg** [comment]
@bjgill I tried `wipe-workspace: true` however, it did not appear to have the intended effect, i.e. the directory still exists e.g. in https://ci.marketplace.team/view/Code%20backups/job/backup-digitalmarketplace-runner/562/console so I have reverted to the previous approach.

**Comment by gidsg** [comment]
@bjgill as discussed, I have force-pushed to provide more details in the commit extended description.

**Review by bjgill — APPROVED** [review]
:shipit:
